src/cmd/flux-keygen.c

```diff
@@ -93,37 +117,34 @@
                 log_msg_exit ("out of memory");
             if ((val = strchr (key, '=')))
                 *val++ = '\0';
-            zcert_set_meta (cert, key, "%s", val ? val : "");
+            meta_set (cert, key, val ? val : "");
             free (key);
         }
     }
-
-    // name is used in overlay logging
-    zcert_set_meta (cert, "name", "%s", hostname);
-    zcert_set_meta (cert, "keygen.hostname", "%s", hostname);
-    zcert_set_meta (cert, "keygen.time", "%s", now);
-    zcert_set_meta (cert, "keygen.userid", "%d", getuid ());
-    zcert_set_meta (cert,
-                    "keygen.flux-core-version",
-                    "%s",
-                    FLUX_CORE_VERSION_STRING);
-    zcert_set_meta (cert,
-                    "keygen.zmq-version",
-                    "%d.%d.%d",
-                    ZMQ_VERSION_MAJOR,
-                    ZMQ_VERSION_MINOR,
-                    ZMQ_VERSION_PATCH);
-    zcert_set_meta (cert,
-                    "keygen.czmq-version",
-                    "%d.%d.%d",
-                    CZMQ_VERSION_MAJOR,
-                    CZMQ_VERSION_MINOR,
-                    CZMQ_VERSION_PATCH);
-
-    if (path && zcert_save_secret (cert, path) < 0)
-        log_msg_exit ("zcert_save_secret %s: %s", path, strerror (errno));
-
-    zcert_destroy (&cert);
+    meta_set (cert, "name", hostname); // used in overlay logging
+    meta_set (cert, "keygen.hostname", hostname);
+    meta_set (cert, "keygen.time", now);
+    meta_set_fmt (cert, "keygen.userid", "%d", getuid ());
+    meta_set (cert, "keygen.flux-core-version", FLUX_CORE_VERSION_STRING);
+    meta_set_fmt (cert,
+                  "keygen.zmq-version",
+                  "%d.%d.%d",
+                  ZMQ_VERSION_MAJOR,
+                  ZMQ_VERSION_MINOR,
+                  ZMQ_VERSION_PATCH);
+
+    if (path) {
+        int fd;
+        FILE *f;
+        if (!(fd = open (path, O_WRONLY | O_CREAT | O_TRUNC, 0600))
```

Comment:
Hmm, I'm not sure I agree with this one.  Basically it wants me to sanitize `path` before opening it for writing in `flux-keygen` but it sounds like they think it's a privileged command or something which it is not.  Any thoughts?

Reply:
Yeah, this seems a little silly. Will the code scanner call out every program that takes an argument to open a file? (e.g. `vi`?) Maybe @dun would have some insight, but I think this could be dismissed as a false positive.

Reply:
I wonder if it would be a good idea to define this cert format in a Flux RFC.  :thinking:

Reply:
If `flux-keygen` is running as the user, I would think this would be equivalent to what they could do at the shell. So I'm not sure what the benefit would be for additional sanitization.

Reply:
Thanks @dun!  OK, I'll dismiss this one then.